### PR TITLE
Secret Manager support for Grafana

### PR DESCRIPTION
**Pull request by garath**
Continuing @greenEkatherine's work from dotnet/arcade-services#1744, this aims to add Secret Manager support for Grafana secrets.

This change fully adopts Secret Manager.

- Create vault-config files for staging and production Grafana instances
- Some secrets were shared between staging and production; separate them as much as reasonable
- Secret Manager has opinions on how some secrets are named; adopt those conventions
- Add Secret Manager checks to Grafana deployment pipeline

This has seen good testing by triggering deployment on a new VM and verifying the configuration comes out correct.

### CODE REVIEW

**Comment by alexperovich** [comment]
Are we sure that every secret in this list is used by grafana? My understanding was that it only used like 2 of them.

**Comment by garath** [comment]
> Are we sure that every secret in this list is used by grafana? My understanding was that it only used like 2 of them.

Yes, they're all used by Grafana. I'm also trying to add some comments on where and how they're used.

**Comment by garath** [comment]
I think I've got this figured out now. This should be close, if not actually, correct to the best settings. I've done a bit of extra work to more completely split "staging" and "production" Grafana, and high-five to Secret Manager for enforcing this better behavior. 

I've also disabled a pile secrets in the Staging keyvault that I know are not in use. 

Next, I'll run the deployment pipelines against a new instance to verify everything is working.

**Comment by garath** [comment]
Related changes needed in helix-service: [!19631](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-service/pullrequest/19631?_a=files)

File: src/Monitoring/grafana-init/grafana.ini [hunk 1]
```diff
@@ -67,7 +67,7 @@ enabled = false
 provider = azure_blob
 
 [external_image_storage.azure_blob]
-account_name = dotnetgrafana
+;account_name = <managed by environment
```
Comment (alexperovich) [inline src/Monitoring/grafana-init/grafana.ini:70]: NIT: missing >

**Review by alexperovich — APPROVED** [review]

**Review by MattGal — APPROVED** [review]

File: .vault-config/shared/dotnet-grafana-secrets.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# API token from https://dotneteng-status{-staging}.azurewebsites.net/
+dotnet-build-bot-dotnet-eng-status-token:
+  type: text
+  parameters:
+    description: Generate API token from https://dotneteng-status{-staging}.azurewebsites.net/
+
+#
+# Was: dotnet-grafana-github-production-oauth-app-secret
```
[the remark this answers is not in the document]
Comment (garath) [reply .vault-config/shared/dotnet-grafana-secrets.yaml:8]: Good note, I'll remove these comments; they were just to help me track the shape of changes.

The story: There are two GitHub OAuth apps: one for staging and one for prod. Their secrets were originally stored in the prod KeyVault with different names. This change moves them to their correct environment KeyVault (prod or staging) but with the same name (so the underlying config always looks for the same secret name, and the keyvault its pointing to is switched at a higher level depending on the environment).

File: .vault-config/shared/dotnet-grafana-secrets.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  type: ad-application
+
+# Grafana Admin user password
+grafana-admin-password:
+  type: text
+  parameters: 
+    description: Log in as a Server Administrator, then use the Users control panel to set the Admin user password.
+
+# Used by Grafana to encrypt data source secrets at-rest; do not rotate lest data become un-decryptable.
+# To rotate, (1) set new key, (2) delete all data sources from Grafana, (3) deploy datasources from version control
```
[the remark this answers is not in the document]
Comment (garath) [reply .vault-config/shared/dotnet-grafana-secrets.yaml:49]: Added a link

**Pull request merged**